### PR TITLE
feat(language): update code example for Cloud NL v2

### PR DESCRIPTION
**Pull request by kornosk**
## Description

Fixes b/296929765

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist

- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Comment by kornosk** [comment]
Hello @minherz , seems like the Kokoro failed due to no v2 available yet. Is this because the new version of the SDK is not not released yet?

File: language/snippets/src/main/java/com/example/language/Analyze.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2016 Google Inc.
+ * Copyright 2023 Google Inc.
```
Comment (minherz) [inline language/snippets/src/main/java/com/example/language/Analyze.java:2]: Please, rollback copyright year. We setup the current year only for new files.

File: language/snippets/src/test/java/com/example/language/AnalyzeIT.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2016 Google Inc.
+ * Copyright 2023 Google Inc.
```
Comment (minherz) [inline language/snippets/src/test/java/com/example/language/AnalyzeIT.java:2]: same here

**Review by minherz — CHANGES_REQUESTED** [review]
LGTM. please, rollback changes in the copyright year.

File: language/snippets/src/main/java/com/example/language/Analyze.java (hunk 1 above)
Comment (kornosk) [reply language/snippets/src/main/java/com/example/language/Analyze.java]: Done.

File: language/snippets/src/test/java/com/example/language/AnalyzeIT.java (hunk 2 above)
Comment (kornosk) [reply language/snippets/src/test/java/com/example/language/AnalyzeIT.java]: Done.

File: language/snippets/src/main/java/com/example/language/Analyze.java [hunk 3]
```diff
@@ -16,27 +16,24 @@
 
 package com.example.language;
 
-import com.google.cloud.language.v1.AnalyzeEntitiesRequest;
-import com.google.cloud.language.v1.AnalyzeEntitiesResponse;
 import com.google.cloud.language.v1.AnalyzeEntitySentimentRequest;
 import com.google.cloud.language.v1.AnalyzeEntitySentimentResponse;
-import com.google.cloud.language.v1.AnalyzeSentimentResponse;
 import com.google.cloud.language.v1.AnalyzeSyntaxRequest;
 import com.google.cloud.language.v1.AnalyzeSyntaxResponse;
```
Comment (minherz) [inline language/snippets/src/main/java/com/example/language/Analyze.java:22]: nit: what is a reason to leave v1 Request / Response imports?

File: language/snippets/src/main/java/com/example/language/Analyze.java [hunk 4]
```diff
@@ -16,27 +16,24 @@
 
 package com.example.language;
 
-import com.google.cloud.language.v1.AnalyzeEntitiesRequest;
-import com.google.cloud.language.v1.AnalyzeEntitiesResponse;
 import com.google.cloud.language.v1.AnalyzeEntitySentimentRequest;
 import com.google.cloud.language.v1.AnalyzeEntitySentimentResponse;
```
Comment (minherz) [inline language/snippets/src/main/java/com/example/language/Analyze.java:20]: nit: what is a reason to leave v1 Request / Response imports?

File: language/snippets/src/main/java/com/example/language/Analyze.java [hunk 5]
```diff
@@ -107,7 +104,6 @@ public static void analyzeEntitiesText(String text) throws Exception {
       // Print the response
       for (Entity entity : response.getEntitiesList()) {
         System.out.printf("Entity: %s", entity.getName());
-        System.out.printf("Salience: %.3f\n", entity.getSalience());
```
Comment (minherz) [inline language/snippets/src/main/java/com/example/language/Analyze.java:110]: nit: is it a removed parameter? Do we want to replace it with smth from v2?

File: language/snippets/src/test/java/com/example/language/AnalyzeIT.java [hunk 6]
```diff
@@ -19,8 +19,8 @@
 import static com.google.common.truth.Truth.assertThat;
 
 import com.google.cloud.language.v1.PartOfSpeech.Tag;
```
Comment (minherz) [inline language/snippets/src/test/java/com/example/language/AnalyzeIT.java:21]: nit: what is a reason to keep v1 import?

**Review by minherz — APPROVED** [review]
LGTM. I left a couple of minor comments. Also I do not see a change to the POM file. Does it mean that the same version of the client library is used to run stuff for v2 API?

File: language/snippets/src/main/java/com/example/language/Analyze.java (hunk 3 above)
Comment (kornosk) [reply language/snippets/src/main/java/com/example/language/Analyze.java:22]: The `analyze_syntax` API is still V1. We need to keep it around until V2 is available.

File: language/snippets/src/main/java/com/example/language/Analyze.java (hunk 4 above)
Comment (kornosk) [reply language/snippets/src/main/java/com/example/language/Analyze.java:20]: The `AnalyzeEntitySentiment` API is still V1. We need to keep it around until V2 is available.

File: language/snippets/src/test/java/com/example/language/AnalyzeIT.java (hunk 6 above)
Comment (kornosk) [reply language/snippets/src/test/java/com/example/language/AnalyzeIT.java:21]: This is for `AnalyzeSyntax` API which is still v1 API. We need to keep it until V2 is available.

File: language/snippets/src/main/java/com/example/language/Analyze.java (hunk 5 above)
Comment (kornosk) [reply language/snippets/src/main/java/com/example/language/Analyze.java:110]: Ahhh, this API is still the `AnalyzeEntitySentiment` v1 API. Updated to use the v1 classes instead. Thanks for catching this!

**Comment by kornosk** [comment]
@minherz Thank you for a quick turnaround! The V2 needs to use a new version of API SDK. Anyway, seems like the SDK will not be published soon. We can wait.

**Comment by kornosk** [comment]
The SDK should be updated around next month.

**Review by minherz — APPROVED** [review]
lgtm

**Review by davidcavazos — APPROVED** [review]
LGTM

**Pull request merged**